demo/site/server.ts

```diff
@@ -53,6 +53,13 @@ app.prepare().then(() => {
                 };
             }
 
+            if (req.headers["rsc"] !== undefined || req.headers["next-router-prefetch"] !== undefined) {
```

Comment:
```suggestion
            if (req.headers["rsc"] !== undefined) {
                // for Rsc requests make sure a _rsc query param exists, to avoid CDN (that ignores Vary header) caching rsc response for a non-rsc request
```